src/components/Alert/Alert.js

```diff
@@ -2,9 +2,16 @@ import React from 'react';
 
 import { StyledAlert } from './Alert.styled';
 
+import { ThemeProvider } from 'styled-components';
+
 const Alert = props => {
+    const {variant} = props
     return (
-        <StyledAlert>{props.children}</StyledAlert>
+        <ThemeProvider theme={ {variant} }>
```

Comment:
Raczej `ThemeProvider` jest wykorzystywany do przekazywania różnych opcji wyglądu.
Czyli jeśli `variant=primary` to "kolory" dla `primary` są pobierane z "theme".
Wtedy `ThemeProvider` powinien być zdefiniowany gdzieś wyżej w strukturze.
Chodzi o to, aby mieć możliwość łatwego i szybkiego przełączania stylów, w tym przypadku będzie to utrudnione.